pkg/networkservice/connectioncontext/dnscontext/client.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dnscontext
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/dnscontext"
+	"github.com/networkservicemesh/sdk/pkg/tools/serialize"
+)
+
+type dnsContextClient struct {
+	cancelMonitoring context.CancelFunc
+	chainContext     context.Context
+	monitorContext   context.Context
+	coreFilePath     string
+	getCallOptions   func() []grpc.CallOption
+	dnsConfigManager dnscontext.Manager
+	monitorClient    networkservice.MonitorConnectionClient
+	executor         serialize.Executor
+}
+
+// NewClient creates a new DNS client chain component. Setups all DNS traffic to the localhost. Monitors DNS configs from connections.
+func NewClient(chainContext context.Context, coreFilePath, resolveConfigPath string, monitorClient networkservice.MonitorConnectionClient, getMonitorCallOptions func() []grpc.CallOption) networkservice.NetworkServiceClient {
```

Comment:
This is a lot of parameters... are there good presumable defaults for many of them?  If so, it might make sense to 
think about something like the WithXYZ pattern we see in grpc.  The goal is to preserve the excellent degree of configuratibility we have, but also make default use simple.  Thoughts?

Reply:
@edwarnicke 

> Thoughts?

Here we are passing `chainContext context.Context`. Probably we can just look up into context values to use specific configuration. For example, if I need to use specific configuration then I can do something like this:
```go
dnscontext.NewClient(dnscontext.WithCorefilePath(ctx, "mydir/Corefile"), monitorClient) 
```
Or If I want to use default values I could just pass context without values
```go
dnscontext.NewClient(ctx, monitorClient) 
```
Do you mind with this pattern?

Reply:
Please use `dnscontext.NewClient(ctx, monitorClient, DnsOption ...options)` I think it would be good.

Reply:
@haiodo , @edwarnicke got it. Thanks, fixed.